src/keycard/keycard.cljs

```diff
@@ -1,38 +1,277 @@
-(ns keycard.keycard)
-
-(defprotocol Keycard
-  (start-nfc [this args])
-  (stop-nfc [this args])
-  (set-nfc-message [this args])
-  (check-nfc-support [this args])
-  (check-nfc-enabled [this args])
-  (open-nfc-settings [this])
-  (register-card-events [this args])
-  (set-pairings [this args])
-  (on-card-disconnected [this callback])
-  (on-card-connected [this callback])
-  (remove-event-listener [this event])
-  (remove-event-listeners [this])
-  (get-application-info [this args])
-  (factory-reset [this args])
-  (install-applet [this args])
-  (install-cash-applet [this args])
-  (init-card [this args])
-  (install-applet-and-init-card [this args])
-  (pair [this args])
-  (generate-and-load-key [this args])
-  (unblock-pin [this args])
-  (verify-pin [this args])
-  (change-pin [this args])
-  (change-puk [this args])
-  (change-pairing [this args])
-  (unpair [this args])
-  (delete [this args])
-  (remove-key [this args])
-  (remove-key-with-unpair [this args])
-  (export-key [this args])
-  (unpair-and-delete [this args])
-  (import-keys [this args])
-  (get-keys [this args])
-  (sign [this args])
-  (sign-typed-data [this args]))
+(ns keycard.keycard
+  (:require
+    ["react-native" :as rn]
+    ["react-native-status-keycard" :default status-keycard]
+    [react-native.platform :as platform]
+    [taoensso.timbre :as log]
+    [utils.address :as address]))
+
+(defonce event-emitter
+  (if platform/ios?
+    (new (.-NativeEventEmitter rn) status-keycard)
+    (.-DeviceEventEmitter rn)))
+
+(defn start-nfc
+  [{:keys [on-success on-failure prompt-message]}]
+  (log/debug "start-nfc")
+  (.. status-keycard
+      (startNFC (str prompt-message))
+      (then on-success)
+      (catch on-failure)))
+
+(defn stop-nfc
+  [{:keys [on-success on-failure error-message]}]
+  (log/debug "stop-nfc")
+  (.. status-keycard
+      (stopNFC (str error-message))
+      (then on-success)
+      (catch on-failure)))
+
+(defn set-nfc-message
+  [{:keys [on-success on-failure status-message]}]
+  (log/debug "set-nfc-message")
+  (.. status-keycard
+      (setNFCMessage (str status-message))
+      (then on-success)
+      (catch on-failure)))
+
+(defn check-nfc-support
+  [{:keys [on-success]}]
+  (.. status-keycard
+      nfcIsSupported
+      (then on-success)))
+
+(defn check-nfc-enabled
+  [{:keys [on-success]}]
+  (.. status-keycard
+      nfcIsEnabled
+      (then on-success)))
+
+(defn open-nfc-settings
+  []
+  (.openNfcSettings status-keycard))
+
+(defn remove-event-listeners
```

Comment:
it's mostly cljs wrapper for `react-native-status-keycard` it's not a part of status-im package, so that's fine to have all API here, but I didn't change much this ns, I just moved it

Reply:
Cool, I would say let's keep an eye on these unused functions. They have negative value because they have to be maintained. In case we end up not using them soon(ish) we can delete them and rely on git.